opentelemetry-sdk/src/opentelemetry/sdk/_metrics/view.py

```diff
@@ -0,0 +1,176 @@
+                the metric stream. If `None`, the name of the instrument will
+                be used.
+
+            description: This is a metric stream customizing attribute: the
+                description of the metric stream. If `None`, the description of
+                the instrument will be used.
+
+            attribute_keys: This is a metric stream customizing attribute: this
+                is a set of attribute keys. If not `None` then only the
+                measurement attributes that are in `attribute_keys` will be
+                used to identify the metric stream.
+
+            aggregation: This is a metric stream customizing attribute: every
+                metric stream has an aggregation instance, this is the class of
+                aggregation this instance will be. If `None` the default
+                aggregation class of the instrument will be used.
+        """
+
+        if (
+            name
+            is instrument_type
+            is instrument_name
+            is meter_name
+            is meter_version
+            is meter_schema_url
+            is None
+        ):
+            raise Exception(
+                "View name or some instrument selection "
+                "criteria must be provided for View {name}"
+            )
+
+        self._instrument_type = instrument_type
+        self._instrument_name = instrument_name
+        self._meter_name = meter_name
+        self._meter_version = meter_version
+        self._meter_schema_url = meter_schema_url
+
+        self._name = name
+
+        if self._name is not None:
+            if self._instrument_type is not None:
+                raise Exception(
+                    "Instrument type matching disabled "
+                    "for view with a defined name"
+                )
+
+            if self._instrument_name is not None:
+                raise Exception(
+                    "Instrument name matching disabled "
+                    "for view with a defined name"
+                )
+
+            if self._meter_name is not None:
+                raise Exception(
+                    "Instrument meter name matching disabled "
+                    "for view with a defined name"
+                )
+
+            if self._meter_version is not None:
+                raise Exception(
+                    "Instrument meter version matching disabled "
+                    "for view with a defined version"
+                )
+
+            if self._meter_schema_url is not None:
+                raise Exception(
+                    "Instrument meter schema URL matching disabled "
+                    "for view with a defined version"
+                )
+
+        self._description = description
+        self._attribute_keys = attribute_keys
+        self._aggregation = aggregation
+
+        self._has_name_and_has_matched = False
+
+    # pylint: disable=too-many-return-statements
+    def _match(self, instrument: Instrument) -> bool:
```

Comment:
Is it intended that users may subclass `View` and override this method to implement custom matching behavior?

Reply:
> it's not officially supported.

can you add the `@final` decorator and maybe make a note saying so in the docstring saying so?

Reply:
Done :v: